lib/strategies/keycloak.js

```diff
@@ -57,8 +57,8 @@ function strategy (config) {
             profile.roles = [];
 
             // Roles are not being properly loaded by the strategy, get them from the access token
-            if (rawProfile.resource_access != null && rawProfile.resource_access.bae != null) {
-                profile.roles = rawProfile.resource_access.bae.roles.map((role) => {
+            if (rawProfile.resource_access != null && rawProfile.resource_access.clientID != null) {
+                profile.roles = rawProfile.resource_access.clientID.roles.map((role) => {
```

Comment:
This is not correct, it had hardcoded bae as client ID within the resource access for the roles. With current solution you are looking for "ClientID" within the resource access. You should look for the configured clientID, something like rawProfile.resource_access[params.clientID]